common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -355,6 +373,30 @@ int addHoverEvent(HoverEvent hoverEvent) {
         return hoverEvents.size();
     }
 
+    public int getWidth() {
+        return McUtils.mc().font.width(getString(PartStyle.StyleType.NONE));
+    }
+
+    public StyledText[] split(String s) {
+        return Arrays.stream(getString().split(s)).map(StyledText::fromString).toArray(StyledText[]::new);
+    }
```

Comment:
This is not even used. Plus, this is a dangerous operation which might not be the way to go about things.

Reply:
This is used in some places i haven't pushed yet. e.g.
```java
private long calculateMessageReadTime(List<StyledText> msg) {
        int words = StyledText.join(" ", msg).split(" ").length;
        long delay =
                dialogAutoProgressDefaultTime.get() + ((long) words * dialogAutoProgressAdditionalTimePerWord.get());
        return delay;
    }
```
or
```java
StyledText[] colored = description.split("\n");
```

Reply:
In these tricky cases just use getString and I will eventually fix them up later in the codebase.